projects/fsastorefrontlib/src/cms-components/checkout/components/configure-product/navigation/product-configuration-navigation.component.ts

```diff
@@ -69,13 +86,16 @@ export class ProductConfigurationNavigationComponent
               1,
               pricingData
             );
-            this.routingService.go({
-              cxRoute: 'addOptions',
-            });
+            if (this.formGroup.valid && this.continueToNextStep) {
```

Comment:
Seems that checking whether form is valid or not is not needed at all since it would be enough to check only which button has been clicked. Or am I missing something?

Reply:
Also if that is the case, perhaps we don't need subject from service at all. Maybe we can resolve everything in this component with the help of that event we are passing on click which holds the information about target taht's been clicked.

Reply:
I tried with event and it doesn't work. Since these components don't have parent/child relationship I have to use `BehaviorSubject` to communicate between them. Also I removed `formGroup`, there is no need for that.